File: cyber/demo_py/tools/demo01_record_write.py
from cyber.python.cyber_py3 import record, cyber
from cyber.demo_base_proto.student_pb2 import Student

if __name__ == "__main__":
    # cyber.init()
    writer = record.RecordWriter()
    writer.set_intervaltime_fileseg(0)
    writer.set_size_fileseg(0)

    record_file = "/apollo/cyber/demo_py/tools/mytest.record"
    if not writer.open(record_file):
        print(f"failed open {record_file}")
        exit(1)
    stu = Student()
    stu.name = "hah"
    stu.height = 1.78
    stu.books.append("cpp")
    stu.books.append("c")
    stu.books.append("python")

    print(f"full name: { stu.DESCRIPTOR.full_name }")
    channel = "chatter"
    
    writer.write_channel("chatter", "cyber.demo_base_proto.Student", "test...")
    for seq in range(20):
        stu.age = seq
        writer.write_message(channel, stu.SerializeToString(), seq)

    writer.close()

File: cyber/demo_cc/tools/demo01_record_write.cc
#include "cyber/cyber.h"
#include "cyber/record/record_writer.h"
#include "cyber/demo_base_proto/student.pb.h"

using apollo::cyber::record::RecordWriter;
using apollo::cyber::demo_base_proto::Student;

int main(int argc, char const *argv[])
{
    apollo::cyber::Init(argv[0]);
    AINFO << "record write demo ...";

    RecordWriter writer;
    writer.SetSizeOfFileSegmentation(0);
    writer.SetIntervalOfFileSegmentation(0);

    writer.Open("/apollo/cyber/demo_cc/tools/mytest.record");
    AINFO << "file : " << writer.GetFile();

    std::string channel = "chatter";
    writer.WriteChannel(channel, "apollo.cyber.demo_base_proto.Student", "test ....");
    
    for(size_t i = 0; i < 1000; i++){
        auto stu = std::make_shared<Student>();
        stu->set_name("huluwa");
        stu->add_books("english");
        stu->add_books("cpp");
        stu->set_age(i);

        std::string content;
        stu->SerializeToString(&content);
        writer.WriteMessage(channel, content, 10000000000000 + i * 10000000);
    }

    writer.Close();
    return 0;
}
